tests/integration/conftest.py

```diff
@@ -662,3 +697,143 @@ def kernel(shell):
 
         # Reboot after clean-up
         shell("tmt-reboot -t 600")
+
+
+@pytest.fixture()
+@pytest.mark.yum_conf_exclude
+def yum_conf_exclude(shell, backup_directory, request):
+    """
+    Fixture.
+    Define `exclude=kernel kernel-core` in /etc/yum.conf.
+    Using pytest.mark.parametrize pass the packages to exclude in a single scenario as a list.
+    Call the fixture directly in the test function.
+    Example:
+        # one testcase
+        @pytest.mark.parametrize("exclude", [["this", "that", "also_this"]])
+
+        # more testcases
+        @pytest.mark.parametrize("exclude", [["this", "that"], ["then_this"]])
+
+        def test_function(yum_conf_exclude, exclude):
+            yum_conf_exclude(exclude)
+    """
+    yum_config = "/etc/yum.conf"
+    backup_dir = backup_directory
+    config = configparser.ConfigParser()
+    config.read(yum_config)
+
+    assert shell(f"cp -v {yum_config} {backup_dir}").returncode == 0
+
+    def _exclude_packages(exclude):
+        for pkg in exclude:
+            # If there is already an `exclude` section, append to the existing value
+            if config.has_option("main", "exclude"):
+                pre_existing_value = config.get("main", "exclude")
+                config.set("main", "exclude", f"{pre_existing_value} {pkg}")
+            else:
+                config.set("main", "exclude", pkg)
+
+        with open(yum_config, "w") as configfile:
+            config.write(configfile, space_around_delimiters=False)
+
+        assert config.has_option("main", "exclude")
+        assert all(pkg in config.get("main", "exclude") for pkg in exclude)
+
+    yield _exclude_packages
+
+    # Clean up
+    assert shell(f"mv {backup_dir}/yum.conf {yum_config}").returncode == 0
+
+
+def _add_client_tools_repo_oracle(shell):
+    """
+    Helper function.
+    Runs only on Oracle Linux system
+    Create an ubi repo for its respective major version to install subscription-manager from.
+    """
+    if "oracle" in SYSTEM_RELEASE_ENV:
+        repo_url = "https://cdn-public.redhat.com/content/public/repofiles/client-tools-for-rhel-8.repo"
+        if SystemInformationRelease.version.major == 7:
+            repo_url = "https://cdn-public.redhat.com/content/public/repofiles/client-tools-for-rhel-7-server.repo"
+
+        # Add the redhat-release GPG key
+        assert shell(f"curl -o /etc/yum.repos.d/client-tools-for-tests.repo {repo_url}")
+        shell("curl -o /etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release https://www.redhat.com/security/data/fd431d51.txt")
```

Comment:
Since this is already being called in another part of the code guarded by the condition of `"oracle" in SYSTEM_RELEASE_ENV`, I would remove the condition from here and keep it simpler for now: 

```suggestion
    repo_url = "https://cdn-public.redhat.com/content/public/repofiles/client-tools-for-rhel-8.repo"
    if SystemInformationRelease.version.major == 7:
        repo_url = "https://cdn-public.redhat.com/content/public/repofiles/client-tools-for-rhel-7-server.repo"

    # Add the redhat-release GPG key
    assert shell(f"curl -o /etc/yum.repos.d/client-tools-for-tests.repo {repo_url}")
    shell("curl -o /etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release https://www.redhat.com/security/data/fd431d51.txt")
```

Reply:
Or de-indent the function call in the places where we are calling them: 

* tests/integration/conftest.py#488
* tests/integration/conftest.py#792